lib/index.ts

```diff
@@ -0,0 +1,132 @@
+var QRCode = require('qrcode')
+
+
+export class VendorMetadata {
+  callbackUrl: URL
+
+  invoiceId: string
+
+  website: URL
+
+  vendorName: string
+
+  static CALLBACK_URL_SHORT = 'cbu'
+
+  static INVOICE_DATA_SHORT = 'ind'
+
+  static WEBSITE_SHORT = 'web'
+
+  static VENDOR_SHORT = 'ven'
+
+  constructor(callbackUrl: URL, invoiceId: string, website: URL, vendorName: string) {
+    this.callbackUrl = callbackUrl
+    this.invoiceId = invoiceId
+    this.website = website
+    this.vendorName = vendorName
+  }
+
+  /**
+   * Converts a [VendorMetadata] object to a concise form for shorter base64 compression
+   *
+   * @returns A concise form of the vendor metadata.
+   */
+  public toConcise(): Object {
+    type Data = {
+        cbu : URL;
+        ind : string;
+        web : URL;
+        ven : string;
+    }
+    let response = {} as Data;
+    response.cbu = this.callbackUrl
+    response.ind = this.invoiceId
+    response.web = this.website
+    response.ven = this.vendorName
+    return response
+  }
+  
+}
+export class PaymentDetails{
+  recipient: string
+
+  amount: number
+
+  reason: string
+
+  category: string
+
+  vendorData?: VendorMetadata
+
+  constructor(recipient: string, amount: number, reason: string, category: string, vendorData: VendorMetadata) {
+    this.recipient = recipient
+    this.amount = amount
+    this.reason = reason
+    this.category = category
+    this.vendorData = vendorData
+  }
+}
+export class TxData{
+  txHash: string
+
+  recipient: string
+
+  amount: number
+
+  invoiceId: string
+
+  category: string
+
+  data?: string
+
+  constructor(txHash: string,recipient: string, amount: number, invoiceId: string, category: string, data: string) {
+    this.txHash = txHash
+    this.recipient = recipient
+    this.amount = amount
+    this.invoiceId = invoiceId
+    this.category = category
+    this.data = data
+  }
+}
+export function generatePaymentLink(PaymentDetails:PaymentDetails) : string {
```

Comment:
Please move each function to its own file, and import them into the index file to expose it as an NPM helper.